core/src/components/checkbox/checkbox.ionic.vars.scss

```diff
@@ -23,6 +23,12 @@ $checkbox-ionic-border-radius: calc(var(--size) / 4 - 2px) !default;
 /// @prop - Icon size of the checkbox for the small size
 $checkbox-ionic-small-size: 16px !default;
 
+/// @prop - Padding top of the checkbox for the default size
+$checkbox-ionic-padding-top: 12px !default;
```

Comment:
Can you share why the ionic theme implementation has padding on the checkbox? This isn't consistent with iOS or MD implementations. 

Checkboxes with this high of padding I would expect are rendered in a list item:
```
<ion-item>
  <ion-checkbox></ion-checkbox>
</ion-item>
```

The `ion-item` would add the respective padding.

Reply:
![image](https://github.com/ionic-team/ionic-framework/assets/60441552/b67f1ae5-b551-4757-8bcf-7914c3b460e0)

I had this padding to ensure that the target area is available for every checkbox.
In the example above I remove the padding and add a background color to the second checkbox :after. 

That's why I add the padding

Reply:
With the padding being added for the target area, I would not expect this diff:

![CleanShot 2024-04-10 at 11 54 08](https://github.com/ionic-team/ionic-framework/assets/13732623/ef468ea3-2c11-498f-a657-ea930dad4d4f)

With the larger checkbox size, it already had the correct minimum target area and should not apply additional padding. 

The target area should operate similar to a "minimum height" and "minimum width" where it will add padding to ensure the checkbox is at least 48px by 48px, but doesn't need to apply past that. 

For example this diff is where you would expect the padding to apply to ensure the target area:

![CleanShot 2024-04-10 at 11 59 06](https://github.com/ionic-team/ionic-framework/assets/13732623/96c8f88e-64c2-4261-8f7f-35010107563d)

If I have some additional time today I can offer some implementation ideas on how to achieve this. Otherwise if you have ideas on how to alter the implementation to achieve this behavior that works too 👍